src/Tribe/Context.php

```diff
@@ -1320,6 +1309,22 @@ public function dangerously_repopulate_locations() {
 		$this->populate_locations();
 	}
 
+	/**
+	 * Just dont...
+	 * Unless you very specifically know what you are doing **DO NOT USE THIS METHOD**!
+	 *
+	 * Please keep in mind this will set force the context to repopulate all locations for the whole request, expensive
+	 * and very dangerous overall since it could affect all this things we hold dear in the request.
+	 *
+	 * With great power comes great responsibility: think a lot before using this.
+	 *
+	 * @since TBD
+	 */
+	public function dangerously_reset_static_properties() {
```

Comment:
Add the `: void` return type?

Reply:
I would update the name to  `dangerously_reset_state`; the `static properties` part suggests **how** it's done; it should suggest the **what**.